### PR TITLE
U14/U18: Every Player has a visual representation of its role, sees other players and whose turn it is

### PR DESCRIPTION
**Pull request by Croxsy**

### CODE REVIEW

**Comment by paul891112** [comment]
The gameplay works fine.

After i leave game, i get redirected to title screen. There, when i create session and click on start, the game screen doesn't load. It only loads after i refresh the page. We might need a cleanup() function that deletes previous game information?

File: src/managers/SessionManager.ts [hunk 1]
```diff
@@ -44,6 +44,15 @@ class SessionManager {
         return session.turnIndex % playerCount === me.orderIndex;
     }
 
+    public isPlayersTurn(player: Player, playerCount: int): boolean {
```
Comment (paul891112) [inline src/managers/SessionManager.ts:47]: Why not just use isMyTurn()?

**Comment by C0DECYCLE** [comment]
(will look at it this weekend)

File: src/scenes/gameuiscreen.ts [hunk 2]
```diff
@@ -100,6 +113,69 @@ export class GameUiScreen extends Phaser.Scene {
         );
     }
 
+    public displayProfiles(): void {
+        const allPlayers: Nullable<Player[]> = PlayerManager.getAll();
```
Comment (paul891112) [inline src/scenes/gameuiscreen.ts:117]: Define a function in PlayerManager that returns the total number of players? I need it in GameUiScreen as well

File: src/scenes/gameuiscreen.ts [hunk 3]
```diff
@@ -100,6 +113,69 @@ export class GameUiScreen extends Phaser.Scene {
...
+        const me: Nullable<Player> = PlayerManager.getMe();
+        const session: Nullable<Session> = SessionManager.get();
+        assert(me && allPlayers && session && this.profilesContainer);
+        this.profilesContainer.removeAll(true);
+        const playerCount: int = allPlayers.length;
```
Comment (paul891112) [inline src/scenes/gameuiscreen.ts:122]: See line 117

File: src/entities/Player.ts
Comment (C0DECYCLE) [inline src/entities/Player.ts:1]: Very good work Rox!

File: src/managers/SessionManager.ts (hunk 1 above)
Comment (C0DECYCLE) [reply src/managers/SessionManager.ts:47]: Would also fuse it with the MyTurn.... the following way:
- Add a optional parameter to isMyTurn (orderIndex).
- Make that the equality check at the end uses the optional parameter or me.index as default.
- Rename function to isTurn.

File: src/scenes/gameuiscreen.ts [hunk 4]
```diff
@@ -100,6 +113,69 @@ export class GameUiScreen extends Phaser.Scene {
...
+                    fontFamily: "Verdana",
+                    fontSize: "11px",
+                    color: "#549b3d",
+                    fontStyle: "bold",
+                },
+            );
+            this.profilesContainer.add(nameText);
+            this.profilesContainer.add(profile);
+        }
+    }
```
Comment (C0DECYCLE) [inline src/scenes/gameuiscreen.ts:177]: I like it but this function is way to big and you should refactor it into multiple smaller ones.

**Review by C0DECYCLE — CHANGES_REQUESTED** [review]
Great work @Croxsy!  Really cool, love the implementation of the feature.

**Review by C0DECYCLE — APPROVED** [review]

**Pull request merged**